rest_framework/utils/field_mapping.py

```diff
@@ -193,7 +193,15 @@ def get_field_kwargs(field_name, model_field):
         ]
 
     if getattr(model_field, 'unique', False):
-        validator = UniqueValidator(queryset=model_field.model._default_manager)
+        unique_error_message = model_field.error_messages.get('unique', None)
+        if unique_error_message:
+            unique_error_message = unique_error_message % {
+                'model_name': model_field.model._meta.object_name,
```

Comment:
I'm not sure what you would like to see. Is it [this](https://github.com/django/django/blob/master/django/db/models/base.py#L1176)?